pkg/lib/interaction/nodes/do_remove_authenticator.go

```diff
@@ -29,11 +31,63 @@ func (n *NodeDoRemoveAuthenticator) Prepare(ctx *interaction.Context, graph *int
 
 func (n *NodeDoRemoveAuthenticator) Apply(perform func(eff interaction.Effect) error, graph *interaction.Graph) error {
 	err := perform(interaction.EffectRun(func(ctx *interaction.Context) error {
-		for _, ai := range n.Authenticators {
-			err := ctx.Authenticators.Delete(ai)
+		userID := graph.MustGetUserID()
+
+		as, err := ctx.Authenticators.List(userID)
+		if err != nil {
+			return err
+		}
+
+		switch n.Authenticator.Kind {
+		case authenticator.KindPrimary:
+			// Ensure all identities have matching primary authenticator.
+			is, err := ctx.Identities.ListByUser(userID)
 			if err != nil {
 				return err
 			}
+
+			aTypes := map[authn.AuthenticatorType]struct{}{}
+			for _, a := range filterAuthenticators(as, authenticator.KeepKind(authenticator.KindPrimary)) {
```

Comment:
Why aren't we using `KeepPrimaryAuthenticatorOfIdentity`?

Reply:
Didn't notice that. Let me change to use it.